ocs_ci/helpers/helpers.py

```diff
@@ -1146,6 +1146,20 @@ def set_image_lookup(image_name):
     return status
 
 
+#  The time and date extraction code below has been modified to read
+#  the month and day data in the logs.  This fixes an error where negative
+#  time values are calculated when test runs cross midnight.  Also, previous
+#  calculations would not set the year, and so the calculations were done
+#  as if the year were 1900.  This is not a problem except that 1900 was
+#  not a leap year and so the next February 29th would throw ValueErrors
+#  for the whole day.  To avoid this problem, changes were made to also
+#  include the current year.
+#
+#  Incorrect times will still be given for tests that cross over from
+#  December 31 to January 1.  Also, there is a lot of code duplication
+#  here that should probably be cleaned up at some point.
```

Comment:
Please create an issue describing the duplication you're talking about here so we can track that.